code/game/objects/items/food/meat.dm

```diff
@@ -14,30 +14,33 @@
 	foodtypes = MEAT
 	w_class = WEIGHT_CLASS_SMALL
 
-/obj/item/food/carpmeat
-	name = "carp fillet"
-	desc = "A fillet of spess carp meat."
+
+/obj/item/food/fishmeat
+	name = "fish fillet"
+	desc = "A fillet of some fish meat."
 	icon_state = "fishfillet"
-	food_reagents = list(/datum/reagent/consumable/nutriment/protein = 4, /datum/reagent/toxin/carpotoxin = 2, /datum/reagent/consumable/nutriment/vitamin = 2)
+	food_reagents = list(/datum/reagent/consumable/nutriment/protein = 4, /datum/reagent/consumable/nutriment/vitamin = 2)
 	bite_consumption = 6
 	tastes = list("fish" = 1)
 	foodtypes = MEAT
 	eatverbs = list("bite","chew","gnaw","swallow","chomp")
 	w_class = WEIGHT_CLASS_SMALL
 
-/obj/item/food/carpmeat/Initialize()
+/obj/item/food/fishmeat/carp
+	name = "carp fillet"
+	desc = "A fillet of spess carp meat."
+	food_reagents = list(/datum/reagent/consumable/nutriment/protein = 4, /datum/reagent/toxin/carpotoxin = 2, /datum/reagent/consumable/nutriment/vitamin = 2)
+	// Cytology category you can swab the meat for.
+	var/cell_line = CELL_LINE_TABLE_CARP
+
+/obj/item/food/fishmeat/carp/Initialize()
 	. = ..()
-	if(!istype(src, /obj/item/food/carpmeat/imitation))
-		AddElement(/datum/element/swabable, CELL_LINE_TABLE_CARP, CELL_VIRUS_TABLE_GENERIC_MOB)
+	AddElement(/datum/element/swabable, cell_line, CELL_VIRUS_TABLE_GENERIC_MOB)
```

Comment:
If `cell_line` is null, does this "do the right thing"?

Reply:
Meant to put this behind null check.